android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java

```diff
@@ -1376,7 +1376,7 @@ private void handleBackgroundImage(KrollDict d)
 					bg, TiConvert.toBoolean(d, TiC.PROPERTY_BACKGROUND_REPEAT, false), bgColor, bgSelected,
 					bgSelectedColor, bgDisabled, bgDisabledColor, bgFocused, bgFocusedColor, gradientDrawable);
 
-				background.setBackgroundDrawable(bgDrawable);
+				nativeView.setBackgroundDrawable(bgDrawable);
```

Comment:
I don't think we should make this change. The `applyCustomBackground()` method already calls `nativeView.setBackgroundDrawable(this.background)`. And the biggest issue with this change is the `handleBackgroundImage()` method (which gets called when background image/color/alpha/etc. changes dynamically) only modifies the `background` member variable and assumes its been assigned to `nativeView`. Your code above replaces the `background` member variable with the `bgDrawable` local drawable reference.

_(Note that I do agree that I don't like how background handling is implemented either. It's overly complicated.)_

Reply:
@ypbnv, I think see why there was a padding issue here.

In the old code, before calling `background.setBackgroundDrawable()`, we call `applyCustomBackground(false)` which in turn calls the main View's `setBackgroundDrawable()` method. If you look at Google's `setBackgroundDrawable()` code in the link below, notice that Google fetches and caches the padding from the background drawable. But the view only fetches the padding when the drawable has been assigned. The view is not aware of any padding changes that are made the drawable after it has been assigned.
https://raw.githubusercontent.com/aosp-mirror/platform_frameworks_base/master/core/java/android/view/View.java

So, the reason the old code's view ignored the padding is because we were setting the 9-patch drawable **after** the background was assigned to the view. This means if we do the below, it should work-around the issue.
```java
background.setBackgroundDrawable(bgDrawable);
if (nativeView != null) {
	nativeView.setBackgroundDrawable(null);
	nativeView.setBackgroundDrawable(background);
}
```

We need to set it `null` first and then reset the background because Google's `View.setBackgroundDrawable()` checks if the reference has changed as an optimization. So, doing the above should fix other potential bugs when changing a view's background dynamically.

Reply:
I tried that, but no effect on the padding problem. It gives me an idea though - to play with the order by which the background is built. Let's see if I can come up with something in that direction.

Reply:
Bummer. I really thought that would do it.
I'm definitely right about the view caching the background drawable's size though.